lib/spack/spack/util/path.py

```diff
@@ -8,37 +8,38 @@
 TODO: this is really part of spack.config. Consolidate it.
 """
 import contextlib
-import errno
 import getpass
 import os
 import re
-import stat
 import subprocess
+import sys
 import tempfile
-from sys import platform as _platform
+
+from six.moves.urllib.parse import urlparse
 
 import llnl.util.tty as tty
-from llnl.util.filesystem import mkdirp
 from llnl.util.lang import memoized
 
-import spack.paths
 import spack.util.spack_yaml as syaml
 
-if _platform == "win32":
-    import win32security
+is_windows = sys.platform == 'win32'
 
 __all__ = [
     'substitute_config_variables',
     'substitute_path_variables',
     'canonicalize_path']
 
+
 # Substitutions to perform
-replacements = {
-    'spack': spack.paths.prefix,
-    'user': getpass.getuser(),
-    'tempdir': tempfile.gettempdir(),
-    'user_cache_path': spack.paths.user_cache_path,
-}
+def replacements():
+    # break circular import from spack.util.executable
```

Comment:
(question) is this still circular? If so, can you post the cycle (the error message includes it IIRC)?

Reply:
Interesting: the comment mentions a cycle involving `executable`, which is not mentioned in this chain (although I do see `util.filesystem` importing `executable` before this PR). Anyhow I might want to look into how to refactor things to avoid this but I don't think that should stop this PR